types/datum.go

```diff
@@ -2283,39 +2284,63 @@ func (ds *datumsSorter) Swap(i, j int) {
 	ds.datums[i], ds.datums[j] = ds.datums[j], ds.datums[i]
 }
 
+var bufPool = &sync.Pool{New: func() interface{} { return new(strings.Builder) }}
```

Comment:
Great!

TiDB uses strings.Builder in many places. but we don't have a pool to reuse it. so i think we can create ```util/sbpool``` to reuse it.

Reply:
OK, I created a new file `utils/pool.go` to put all pools...